### DISCUSSION TITLE
Git repository: Refusing to update because there are uncommited changes.

### DISCUSSION THREAD

**Discussion by mxxivcz** [orgzly-revived/orgzly-android-revived#352] [General]
Hi, I'm testing git repository to synchronise the content. I came to a state where synchronisation is not possible, as the error mentioned in the title appears. Is there a way to get rid of this? Ideally other than removing the repository and adding it again :). 

I was not adding or renaming org notebooks, just changing existing and syncing.

**Comment by amberin** [comment]
Interesting. It should be impossible to reach this state, and I've never seen it happen before... I'm very curious where the bug is.... 🤔

I assume you have not touched the Git working directory outside of Orgzly?

Have you tried force-save or force-load? That's where I would start. (Select the notebook, and two cloud buttons appear.)

**Reply by mxxivcz** [reply]
I didn't touch git on Android outside Orgzly. I was only syncing changes from laptop. I already had ~100 org files to sync. I also had some org files in a subfolder and I removed that subfolder from the git (from laptop). 

Force-save or force-load is not possible, I have the same error message.

**Reply by amberin** [reply]
Hm, if there is a subfolder involved, the bug could be related to that. This type of error didn't use to happen at all.

I'm afraid your best bet is to remove the repository. You can try leaving Orgzly's books as they are. If we are lucky, they might find their "namesakes" without issues when you add the repo again.

**Reply by mxxivcz** [reply]
It was actually a good hint when you asked if I touched the repo outside Orgzly :). I installed termux and checked the repo. 

The repo is currently in the branch merge-SomeFile.org-2382983-date. There are a couple of modified files and some deleted files from the subfolder I removed.

**Reply by amberin** [reply]
@mxxivcz The fact that the workdir has that branch checked out indicates that there may have been a sync conflict, but not necessarily.

I'm very sorry, this is the first time I hear of the Git merge handling leading to this kind of mess. I will try to reproduce this to find the issue.

Did you have subfolder support enabled or disabled (the default) when you synced the changes from your laptop?

**Reply by mxxivcz** [reply]
I didn't change any preference about subfolder support, it must have stayed with the default value. I didn't see the org files from the subfolder in Orgzly (I have basic org mode files in the top-level folder, then I have Roam folder nested in it). 

I did a copy of that stuck git folder from the command line in termux and resolved the issue, so I'm fine now. It's good to know that I can any time jump to shell and check what's going on there.

**Reply by amberin** [reply]
Thank you, that's useful information.

Yes, it's good for you in this case, but ideally the git workdir should be hidden in private storage, that's what we're aiming for. But then it obviously needs to be robust enough that these situations do not arise.

Another question: If you look at the Git repository settings, does the specified main branch match your actual main branch name? Because there is a bug here, and it could be related.

**Reply by mxxivcz** [reply]
>  If you look at the Git repository settings, does the specified main branch match your actual main branch name? 

I used main as a branch name everywhere.
